workers/mocklogger_test.go

```diff
@@ -0,0 +1,46 @@
+package workers
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/cryptopay-dev/yaga/logger/nop"
+)
+
+type mockLogger struct {
+	*nop.Logger
+
+	mu       *sync.Mutex
+	messages []string
+}
+
+func newMockLogger() *mockLogger {
+	return &mockLogger{
+		Logger: new(nop.Logger),
+		mu:     new(sync.Mutex),
+	}
+}
+
+func (l *mockLogger) Error(i ...interface{}) {
+	l.mu.Lock()
+	defer l.mu.Unlock()
+	l.messages = append(l.messages, fmt.Sprint(i...))
```

Comment:
can be make a count of errors, and then check that there was one error? Instead of comparing the text that can change
может быть сделать счетчик errors, и затем проверить что была одна ошибка? а не сравнивать текст который может поменяться

Reply:
согласен, можно так думал, но на счётчик как то не доверяю ориентироваться

Reply:
check for string content even worse
you should have only one error, or if there are more of them or no one, this is already wrong behavior

Reply:
ok

Reply:
хммм, мы можем получить скрытую подмену ошибки (поведения). Например, потом изменяем внутри реализацию воркеров, и вместо одной ошибки туда будет попадать другая ошибка из другого места. Так как мы в тестах проверяем кол-во, то не сработает тест на определенную ошибку поведения.

Честно и так и так не очень, предлагаю проголосовать ))
* проверять текст
* проверять кол-во

Reply:
Check text of error is bad practice